util/HalideTraceUtils.cpp

```diff
@@ -0,0 +1,52 @@
+#include "HalideTraceUtils.h"
+#include <stdint.h>
+#include <assert.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+bool Packet::read_from_stdin() {
+    return read_from_filedesc(0);
+}
+
+bool Packet::read_from_filedesc(int fdesc){
+    uint32_t header_size = (uint32_t)sizeof(halide_trace_packet_t);
+    if (!Packet::read(this, header_size, fdesc)) {
+        return false;
+    }
+    uint32_t payload_size = size - header_size;
```

Comment:
Why not just use size_t here instead of casting everything to uint32? Packet::size will upcast to size_t automatically.